### PR TITLE
Security additions + edits

### PR DESCRIPTION
**Pull request by nemild**
With changes in Ethereum and Solidity best practice, I've updated the doc to highlight security risk - and fix errors and/or bad security patterns in the previous doc

### CODE REVIEW

File: solidity.html.markdown [hunk 1]
```diff
@@ -409,6 +430,21 @@ modifier checkValue(uint amount) {
...
 // to boolean (comparison operators must be used to get the boolean val)
 
+// For loops that are determined by user behavior, be careful - as contracts have a maximal
+// amount of gas for a block of code - and will fail if that is exceeded
+// For example:
+for( uint x = 0; x < refundAddressList.length; x++ ) {
+  if (!refundAddressList[x].send(SOME_AMOUNT)) {
+     throw;
+  }
+  }
```
Comment (vendethiel) [inline solidity.html.markdown]: indentation

File: solidity.html.markdown [hunk 2]
```diff
@@ -409,6 +430,21 @@ modifier checkValue(uint amount) {
 // Syntax same as javascript, but no type conversion from non-boolean
 // to boolean (comparison operators must be used to get the boolean val)
 
+// For loops that are determined by user behavior, be careful - as contracts have a maximal
+// amount of gas for a block of code - and will fail if that is exceeded
+// For example:
+for( uint x = 0; x < refundAddressList.length; x++ ) {
```
Comment (vendethiel) [inline solidity.html.markdown]: spacing

**Comment by vendethiel** [comment]
Why close? :o

**Comment by vendethiel** [comment]
Oh, nice. Can I merge now, or do you still want to work on fleshing out some things?

**Comment by nemild** [comment]
Sorry, that was a mistake - thanks for the quick review. Nope, it's good for now.

**Comment by vendethiel** [comment]
Thanks! 👍

**Pull request merged**